inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -168,16 +205,39 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(c
     }
 
     auto fullConfig = mergeConfigs(_config, config);
+    // collect the settings that are applicable to the devices we are loading the network to
+    std::unordered_map<std::string, InferenceEngine::Parameter> multiNetworkConfig;
+    std::vector<DeviceInformation> metaDevices;
+    auto workMode = fullConfig.find(PluginConfigParams::KEY_WORK_MODE);
     auto priorities = fullConfig.find(MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES);
+
+    // not found device priorities for -d AUTO use case
     if (priorities == fullConfig.end()) {
-        IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for MULTI device";
+        if (workMode != fullConfig.end() && workMode->second == PluginConfigParams::AUTO) {
+            std::string allDevices;
+            auto availableDevices = GetCore()->GetAvailableDevices();
+            if (availableDevices.empty()) {
+                IE_THROW(NotFound) << "No available device found";
+            }
+            for (auto&& device : availableDevices) {
+                allDevices += device;
+                allDevices += ((device == availableDevices[availableDevices.size()-1]) ? "" : ",");
+            }
+            metaDevices = ParseMetaDevices(allDevices, fullConfig);
+            multiNetworkConfig.insert({MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, allDevices});
+        } else {
+            IE_THROW() << "KEY_MULTI_DEVICE_PRIORITIES key is not set for MULTI device";
```

Comment:
this else is for if(workMode->second == PluginConfigParams::AUTO), are you sure we have to throw on having no DEVICE_PRIORITIES?